pkg/p2p/streamtest/streamtest_test.go

```diff
@@ -314,6 +314,72 @@ func TestRecorder_closeAfterPartialWrite(t *testing.T) {
 			return err
 		}
 
+		// stream should be closed and write should return err
+		if _, err := rw.WriteString("expect err message"); err != nil {
+			return fmt.Errorf("write: %w", err)
+		}
+
+		if err := rw.Flush(); err == nil {
+			return fmt.Errorf("expected err")
+		}
```

Comment:
also, maybe we can test that `Read` still works (since it should after a `Close`)

Reply:
I agree, I will also add both read and write checks after reset as well, in the `TestRecorder_resetAfterPartialWrite`